scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1290,6 +1290,62 @@
             (catch Exception e
               (log/error e "Unable to decline offers!")))))))
 
+(defn create-mesos-scheduler
+  "Creates the mesos scheduler which offloads processing of status updates to an agent.
+   The agent is used to ensure in-order asynchronous processing of status updates."
```

Comment:
nitpick: put this line as a comment above the `processor-agent`

Larger comment though is that since we are just using the agent as in order async processing perhaps it is worth making a function / interface that makes that clear and then the implementation is the agent. This way, the thing we are trying to achieve is upfront.